lib/Digitick/Sepa/PaymentInformation.php

```diff
@@ -165,17 +165,17 @@ class PaymentInformation
     /**
      * @param string $id
      * @param string $originAccountIBAN This is your IBAN
-     * @param string $originAgentBIC This is your BIC
      * @param string $originName This is your Name
-     * @param string $originAccountCurrency
+     * @param string $originAccountCurrency The HCT defaults to HUF
+     * @param string $originAgentBIC Not required for HCT
      */
-    public function __construct($id, $originAccountIBAN, $originAgentBIC, $originName, $originAccountCurrency = 'EUR')
+    public function __construct($id, $originAccountIBAN, $originName, $originAccountCurrency = 'HUF', $originAgentBIC = null)
```

Comment:
As the one above, this introduces a BC break and can be circumvented by a corresponding solution in the builder.
The whole system is created on the visitor pattern and thus, with another visitor for HUF we can beat most of your requirements.

Reply:
I based the change on HUF as the currency code, but the better naming would be HCT to escape any confusion on currency vs diverted functionality

Reply:
Still I don't have an idea where, and how to provide the info that we are going to build the xml on HCT-way, if i understand right, how the factory code works, one elegant way would be to introduce the copies of the classes that relate to CustomerCreditTransfer* but named CustomerHUngarianCreditTransfer* 

But here i think that hungary would get too much "attention" and maintain the code with that extra bunch of classes, would be out of scope, as the SEPA is the core objective, and any EUR transfers to and from hungary are also SEPA aynway.

That is where i get back to the initial question, if i fork off the project and maintain the separate codebase using the upstream changes, also ensure the differentiation to "digitick/sepa-xml" in naming, your "part" would be done. ;)

Reply:
As I worte in my comment I'm gogin to think about a good BC break free solution to this, The visitor pattern is build exactly for this case, and together with a strategy or similar we can subtitute the whole xml generation process to conform with you requirements, and splitting this into a separate package is totally ok for me, I want to take away from you to have to maintain a slightly changed copy of the sepa library as this will introduce a lot of work for you. And if an "adapater" package for HCT is just depending on the sepa xml introducing only the code used to change some minor behaviour this would be much easier to maintain for you.

Reply:
Yes, at the beginning i modified the code in ./vendor/ to get my task done, but later i was thinking about to provide back my changes if they may be useful for others, and was thinking about what i read/learned on class inheritance, and extending, but had no idea how to start as i am in no means a real programmer :D

Reply:
I think you can open the PR to allow to push me into it, I would then scribble my ideas directly here

Reply:
**Allow edits by maintainers** is ticked so youre free to commit as i understand